src/Database/Connection.php

```diff
@@ -869,6 +810,9 @@ public function isQueryLoggingEnabled(): bool
     public function setLogger(LoggerInterface $logger): void
     {
         $this->_logger = $logger;
+        if ($this->_logQueries) {
+            $this->_driver->setLogger($this->_logger);
+        }
```

Comment:
Should we have logger on both connection and Driver? In the places we use logging in the connection couldn't we get the logger from the Driver?